src/business_logic/transaction/internal_objects.rs

```diff
@@ -166,3 +170,246 @@ impl InternalDeploy {
         todo!()
     }
 }
+
+#[derive(Clone, Debug)]
+pub struct InternalDeployAccount {
+    contract_address: Address,
+    contract_address_salt: Address,
+    class_hash: [u8; 32],
+    constructor_calldata: Vec<Felt>,
+    version: u64,
+    nonce: u64,
+    max_fee: u64,
+    signature: Vec<Felt>,
+    chain_id: StarknetChainId,
+}
+
+impl InternalDeployAccount {
+    #[allow(clippy::too_many_arguments)]
+    pub fn new(
+        class_hash: [u8; 32],
+        max_fee: u64,
+        version: u64,
+        nonce: u64,
+        constructor_calldata: Vec<Felt>,
+        signature: Vec<Felt>,
+        contract_address_salt: Address,
+        chain_id: StarknetChainId,
+    ) -> Result<Self, SyscallHandlerError> {
+        let contract_address = calculate_contract_address(
+            &contract_address_salt,
+            &Felt::from_bytes_be(&class_hash),
+            &constructor_calldata,
+            Address(Felt::zero()),
+        )?;
+
+        Ok(Self {
+            contract_address: Address(contract_address),
+            contract_address_salt,
+            class_hash,
+            constructor_calldata,
+            version,
+            nonce,
+            max_fee,
+            signature,
+            chain_id,
+        })
+    }
+
+    pub fn new_for_testing(
+        contract_class: &ContractClass,
+        max_fee: u64,
+        contract_address_salt: Address,
+        constructor_calldata: Vec<Felt>,
+        chain_id: StarknetChainId,
+        signature: Vec<Felt>,
+    ) -> Result<Self, SyscallHandlerError> {
+        Self::new(
+            felt_to_hash(&compute_class_hash(contract_class).unwrap()),
+            max_fee,
+            TRANSACTION_VERSION,
+            0,
+            constructor_calldata,
+            signature,
+            contract_address_salt,
+            chain_id,
+        )
+    }
+
+    pub fn get_state_selector(&self, _general_config: StarknetGeneralConfig) -> StateSelector {
```

Comment:
we have not add the selector function in any of the others internal transactions. It would be cool to check if it is actually necessary.